sdk/web-pubsub/web-pubsub-express/src/cloudEventsDispatcher.ts

```diff
@@ -136,10 +162,129 @@ function tryGetWebPubSubEvent(req: IncomingMessage): EventType | undefined {
+      case MqttV500ConnectReasonCode.TopicNameInvalid:
+      case MqttV500ConnectReasonCode.PayloadFormatInvalid:
+      case MqttV500ConnectReasonCode.ImplementationSpecificError:
+      case MqttV500ConnectReasonCode.PacketTooLarge:
+      case MqttV500ConnectReasonCode.RetainNotSupported:
+      case MqttV500ConnectReasonCode.QosNotSupported:
+        return 400; // BadRequest
+      case MqttV500ConnectReasonCode.QuotaExceeded:
+      case MqttV500ConnectReasonCode.ConnectionRateExceeded:
+        return 429; // TooManyRequests
+      case MqttV500ConnectReasonCode.Banned:
+        return 403; // Forbidden
+      case MqttV500ConnectReasonCode.UseAnotherServer:
+      case MqttV500ConnectReasonCode.ServerMoved:
+      case MqttV500ConnectReasonCode.ServerUnavailable:
+      case MqttV500ConnectReasonCode.ServerBusy:
+      case MqttV500ConnectReasonCode.UnspecifiedError:
+        return 500; // InternalServerError
+      default:
+        logger.warning(`Invalid MQTT connect return code: ${mqttConnectCode}.`);
+        return 500; // InternalServerError
+    }
+  }
+}
+
+function getMqttConnectCodeFromStatusCode(
+  statusCode: 400 | 401 | 500,
+  protocolVersion: number,
+): MqttV311ConnectReturnCode | MqttV500ConnectReasonCode {
+  if (protocolVersion === 4) {
+    switch (statusCode) {
+      case 400:
+        return MqttV311ConnectReturnCode.BadUsernameOrPassword;
+      case 401:
+        return MqttV311ConnectReturnCode.NotAuthorized;
+      case 500:
+        return MqttV311ConnectReturnCode.ServerUnavailable;
+      default:
+        logger.warning(`Unsupported HTTP Status Code: ${statusCode}.`);
+        return MqttV311ConnectReturnCode.ServerUnavailable;
+    }
+  } else if (protocolVersion === 5) {
+    switch (statusCode) {
+      case 400:
+        return MqttV500ConnectReasonCode.BadUserNameOrPassword;
+      case 401:
+        return MqttV500ConnectReasonCode.NotAuthorized;
+      case 500:
+        return MqttV500ConnectReasonCode.UnspecifiedError;
+      default:
+        logger.warning(`Unsupported HTTP Status Code: ${statusCode}.`);
+        return MqttV500ConnectReasonCode.UnspecifiedError;
+    }
+  } else {
+    logger.warning(`Invalid MQTT protocol version: ${protocolVersion}.`);
+    return MqttV311ConnectReturnCode.UnacceptableProtocolVersion;
+  }
+}
+
+function handleConnectErrorResponse(
+  connectRequest: ConnectRequest,
+  response: ServerResponse,
+  code: 400 | 401 | 500,
+  detail?: string,
+): void {
+  const isMqttReq = connectRequest.context.clientProtocol === "mqtt";
+  if (isMqttReq) {
+    console.log("mqtt req ", connectRequest);
+    const protocolVersion = (connectRequest as MqttConnectRequest).mqtt.protocolVersion;
+    const mqttErrorResponse: MqttConnectErrorResponse = {
+      mqtt: {
+        code: getMqttConnectCodeFromStatusCode(code, protocolVersion),
+        reason: detail,
+      },
+    };
+    response.statusCode = code;
+    response.end(JSON.stringify(mqttErrorResponse));
```

Comment:
also here, is json content type needed?

Reply:
yes it is needed to match the spec here
<img width="458" alt="image" src="https://github.com/user-attachments/assets/118be085-e20b-4857-bdfd-6363024949cf">